src/main/resources/static/js/app.js

```diff
@@ -0,0 +1,119 @@
+var app = angular.module("JournalApp", []);
+
+app.controller("CategoryController", function ($scope, $http) {
+
+    $scope.isAnySubscription = function() {
+        var result = false;
+        for(var i=0; i< $scope.subscriptions.length; i++) {
+            if($scope.subscriptions[i].active) {
+                return true;
+            }
+        }
+        return result;
+    }
+
+    $scope.subscribe = function (categoryId) {
+        $http.post('/rest/journals/subscribe/' + categoryId).success(function (data, status, headers, config) {
+            for (var i = 0; i < $scope.subscriptions.length; i++) {
+                var s = $scope.subscriptions[i];
+                if (s.id == categoryId) {
+                    $scope.subscriptions[i].active = true;
+                    break;
+                }
+            }
+        }).error(function (data, status, headers, config) {
+            if (data.message == 'Time is out') {
+                console.log("Timeout")
+            }
+        });
+    }
+
+    $scope.getCategories = function () {
+        $http.get('/public/rest/category').success(function (data, status, headers, config) {
+            $scope.categories = data;
+        }).error(function (data, status, headers, config) {
+            if (data.message == 'Time is out') {
+                console.log("Timeout")
+            }
+        });
+    }
+
+    $scope.getSubscriptions = function () {
+        $http.get('/rest/journals/subscriptions').success(function (data, status, headers, config) {
+            $scope.subscriptions = data;
+        }).error(function (data, status, headers, config) {
+            if (data.message == 'Time is out') {
+                console.log("Timeout")
+            }
+        });
+    }
+});
+
+app.controller("UserSubscriptionController", function ($scope, $http) {
+    $scope.getCategories = function () {
+        $http.get('/public/rest/category').success(function (data, status, headers, config) {
+            $scope.categories = data;
+        }).error(function (data, status, headers, config) {
+            if (data.message == 'Time is out') {
+                console.log("Timeout")
+            }
+        });
+    }
+});
+
+app.controller("JournalController", function ($scope, $http, $filter) {
+    $scope.getJournals = function () {
+        $http.get('/rest/journals').success(function (data, status, headers, config) {
```

Comment:
`$http` functions are pretty much confirmant to ES2015's Promise A+ spec. I would recommend to follow the spec instead.